src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -1005,6 +1005,34 @@ public void SendIovMaxUdp_SuccessOrMessageSize()
             Assert.True(receiveTask.IsCompleted);
             await receiveTask;
         }
+
+        [Fact]
+        public void SocketSendWouldBlock_ReturnsBytesSent()
+        {
+            using (var server = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            using (var client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                // listen
+                server.BindToAnonymousPort(IPAddress.Loopback);
+                server.Listen(1);
+                // connect
+                client.Connect(server.LocalEndPoint);
+                // accept
+                using (Socket socket = server.Accept())
+                {
+                    // We send a large amount of data but don't read it.
+                    // A chunck will be sent, attempts to send more will return SocketError.WouldBlock.
+                    // Socket.Send must return the success of the partial send.
+                    socket.Blocking = false;
+                    var data = new byte[100_000_000];
```

Comment:
This is not the kind of test we should run in CI and not for innerloop.  Allocating this kind of memory makes this test a stress test and not a unit/functionaltest.

Reply:
The amount of memory could be lowered.  This isn't a stress test; it's just picking a value that won't be completed in a single write because it's larger than the OS' buffer.

Reply:
On my Linux machine this sends 2772174 bytes. I'll change this to 5M and see what CI thinks of it.